src/js/src/components/SearchBox.vue

```diff
@@ -118,14 +134,26 @@ export default {
           this.futureUrlSearch !== this.solrSettings.urlSearch ||
           this.futureImgSearch !== this.solrSettings.imgSearch) 
         {
-        console.log('search fired!')
+        //console.log('search params changed!')
+        this.clearResults()
         this.updateQuery(this.futureQuery)
         this.futureGrouped = this.solrSettings.grouping
         this.futureUrlSearch = this.solrSettings.urlSearch
         this.futureImgSearch = this.solrSettings.imgSearch
-        this.requestSearch({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-        this.requestFacets({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-        this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
+        if(this.solrSettings.imgSearch === true) {
+           this.requestImageSearch({query:this.query})
+           this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
+          return
+        }
+        if(this.solrSettings.urlSearch === true) {
+          return
+        }
+        else {
+          this.requestSearch({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
+          this.requestFacets({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
+          this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
+          return
+        }
```

Comment:
I see how this definitely could use some DRY'ing out now ;)

Reply:
Yea, I've made a jira JUST for that!